pkg/pool-manager/pod_pool.go

```diff
@@ -61,7 +61,10 @@ func (p *PoolManager) AllocatePodMac(pod *corev1.Pod) error {
 	}
 
 	// validate if the pod is related to kubevirt
-	if p.isRelatedToKubevirt(pod) {
+	// since the pod request may not have owned the vm namespace in this early state of vm making,
+	// we need to receive the namespace from the network annotation, and because the namespace is
+	// the same for all the networks, we simply take the namespace from the first network
```

Comment:
Can we add a check here that ensure that all Namespaces are the same ? so we are sure that taking first one is good ? that's the best way to document it.

Maybe at a `selectNamespaceFromNetworks` function so we can put the logic there.